### PR TITLE
fix and simulate engine disconnect when in sketch mode

### PR DESCRIPTION
**Pull request by jessfraz**
closes https://github.com/KittyCAD/modeling-app/issues/584 and adds a test
ya'll can make this fancier in the future but this at least prevents bad things from happening and shows a user can continue sketching after

### CODE REVIEW

File: src/components/ModelingMachineProvider.tsx [hunk 1]
```diff
@@ -658,6 +662,19 @@ export const ModelingMachineProvider = ({
     editorManager.selectionRanges = modelingState.context.selectionRanges
   }, [modelingState.context.selectionRanges])
 
+  useEffect(() => {
+    const offlineCallback = () => {
+      // If we are in sketch mode we need to exit it.
+      // TODO: how do i check if we are in a sketch mode, I only want to call
```
Comment (jessfraz) [inline src/components/ModelingMachineProvider.tsx:668]: @Irev-Dev @franknoirot I feel like there is a way to do this so just let me know

File: e2e/playwright/test-utils.ts [hunk 2]
```diff
@@ -196,7 +196,7 @@ export async function getUtils(page: Page) {
     ) => {
       // Skip on non-Chromium browsers, since we need to use the CDP.
       test.skip(
-        cdpSession === null,
+        browserType !== 'chromium',
```
Comment (jessfraz) [inline e2e/playwright/test-utils.ts:199]:
```suggestion
        cdpSession === null,
```

File: src/components/ModelingMachineProvider.tsx (hunk 1 above)
Comment (jessfraz) [reply src/components/ModelingMachineProvider.tsx:668]: on second thought if we do this the useeffect will refresh on state changes and i dont think thats worth it

Comment (Irev-Dev) [reply src/components/ModelingMachineProvider.tsx:668]: `state.matches('Sketch')`

**Pull request merged**